code/modules/mob/living/basic/space_fauna/carp/carp.dm

```diff
@@ -119,6 +120,10 @@
 	ai_controller.set_blackboard_key(BB_CARP_RIFT, teleport)
 	ai_controller.set_blackboard_key(BB_OBSTACLE_TARGETING_WHITELIST, allowed_obstacle_targets)
 
+/mob/living/basic/carp/Destroy()
+	QDEL_NULL(teleport)
```

Comment:
If anyone could tell me why not including this causes a hard deletion now but didn't before I added the speechmod component I would be appreciative it's a total mystery to me